plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java

```diff
@@ -0,0 +1,298 @@
+                .hasMessageContaining("External location is not a valid file system URI")
+                .hasStackTraceContaining("Fragment is not allowed in a file system location");
+    }
+
+    @Test
+    public void testCtasWithIncorrectLocation()
+    {
+        String tableName = "test_ctas_with_incorrect_location_" + randomNameSuffix();
+        String location = "s3://%s/%s/a#hash/%s".formatted(bucketName, schemaName, tableName);
+
+        assertThatThrownBy(() -> assertUpdate("CREATE TABLE " + tableName + "(col_str, col_int)" +
+                " WITH (external_location = '" + location + "')" +
+                " AS VALUES ('str1', 1)"))
+                .hasMessageContaining("External location is not a valid file system URI")
+                .hasStackTraceContaining("Fragment is not allowed in a file system location");
+    }
+
+    @Test
+    public void testCreateSchemaWithIncorrectLocation()
+    {
+        String schemaName = "test_create_schema_with_incorrect_location_" + randomNameSuffix();
+        String schemaLocation = "s3://%s/%2$s/a#hash/%2$s".formatted(bucketName, schemaName);
+        String tableName = "test_basic_operations_table_" + randomNameSuffix();
+        String qualifiedTableName = schemaName + "." + tableName;
+
+        assertUpdate("CREATE SCHEMA " + schemaName + " WITH (location = '" + schemaLocation + "')");
+        assertThat(getSchemaLocation(schemaName)).isEqualTo(schemaLocation);
+
+        assertThatThrownBy(() -> assertUpdate("CREATE TABLE " + qualifiedTableName + "(col_str, col_int) AS VALUES ('str1', 1)"))
+                .hasMessageContaining("Fragment is not allowed in a file system location");
+
+        assertThatThrownBy(() -> assertUpdate("CREATE TABLE " + qualifiedTableName + "(col_str varchar, col_int integer)"))
+                .hasMessageContaining("Fragment is not allowed in a file system location");
+
+        assertUpdate("DROP SCHEMA " + schemaName);
+    }
+
+    @Test
+    public void testSchemaNameEscape()
+    {
+        String schemaNameSuffix = randomNameSuffix();
+        String schemaName = "../test_create_schema_escaped_" + schemaNameSuffix;
+        String tableName = "test_table_schema_escaped_" + randomNameSuffix();
+
+        assertUpdate("CREATE SCHEMA \"%2$s\" WITH (location = 's3://%1$s/%2$s')".formatted(bucketName, schemaName));
+        assertQueryFails("CREATE TABLE \"" + schemaName + "\"." + tableName + " (col) AS VALUES 1", "Failed checking path: .*");
+
+        assertUpdate("DROP SCHEMA \"" + schemaName + "\"");
+    }
+
+    @Override
+    protected List<String> locationPatterns()
+    {
+        // TODO https://github.com/trinodb/trino/issues/17803 Fix correctness issue with double slash
+        return super.locationPatterns().stream()
+                .filter(location -> !location.contains("double_slash"))
```

Comment:
This is extracted from https://github.com/trinodb/trino/pull/17804
is this the only place that's different?

Reply:
This PR doesn't contain `adaptTableLocation`, `testCreateTableWithDoubleSlash` and `testCtasWithDoubleSlash`.
* `adaptTableLocation` is an extension point when we run the test with HMS in the future. 
* `testCreateTableWithDoubleSlash` and `testCtasWithDoubleSlash` require some changes in `HiveLocationService`

Reply:
> * `testCreateTableWithDoubleSlash` and `testCtasWithDoubleSlash` require some changes in `HiveLocationService`

or reverting de-hadooping of HiveLocationService?